src/utils/parsers.rs

```diff
@@ -20,6 +22,62 @@ pub fn parse_schema_source(loc: &str) -> Result<SchemaSource> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct GraphRef {
+    pub name: String,
+    pub variant: String,
+}
+
+/// NOTE: THIS IS A TEMPORARY SOLUTION. IN THE FUTURE, ALL GRAPH ID PARSING
+/// WILL HAPPEN IN THE BACKEND TO KEEP EVERYTHING CONSISTENT. THIS IS AN
+/// INCOMPLETE PLACEHOLDER, AND MAY NOT COVER EVERY SINGLE VALID USE CASE
+pub fn parse_graph_ref(graph_id: &str) -> Result<GraphRef> {
+    let pattern = Regex::new(r"^[a-zA-Z][a-zA-Z0-9_-]{0,63}$").unwrap();
+    let variant_pattern = Regex::new(r"^([a-zA-Z][a-zA-Z0-9_-]{0,63})@(.{0,63})$").unwrap();
+
+    let valid_graph_name_only = pattern.is_match(graph_id);
+    let valid_graph_with_variant = variant_pattern.is_match(graph_id);
+
+    if valid_graph_name_only {
+        Ok(GraphRef {
+            name: String::from(graph_id),
+            variant: String::from("current"),
+        })
+    } else if valid_graph_with_variant {
+        let matches = variant_pattern.captures(graph_id).unwrap();
+        let name = matches.get(1).unwrap().as_str();
+        let variant = matches.get(2).unwrap().as_str();
+        Ok(GraphRef {
+            name: String::from(name),
+            variant: String::from(variant),
+        })
+    } else {
+        Err(anyhow!("Graph IDs must be in the format <NAME> or <NAME>@<VARIANT>, where <NAME> can only contain letters, numbers, or the characters `-` or `_`, and must be 64 characters or less. <VARIANT> must al be 64 characters or less."))
```

Comment:
This error is mighty long, but without validating the name/variant separately, I think it's the best I can do. I think it'll be fine for now, since it's gonna be stripped out soon. Thoughts?

Reply:
lgtm - but `al` should be removed